pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -513,7 +513,6 @@ func (m *otelMetricsOperatorInstance) shutdown() {
 }
 
 func (m *otelMetricsOperatorInstance) init(gadgetCtx operators.GadgetContext) error {
```

Comment:
I know the goal of this PR is different, but what about this alternative implementation of this function: Is it equivalent?

```go
func (m *otelMetricsOperatorInstance) init(gadgetCtx operators.GadgetContext) error {
	for _, ds := range gadgetCtx.GetDataSources() {
		annotations := ds.Annotations()
		metricsCollect := annotations[AnnotationMetricsCollect] == "true"
		// We only allow printing if the gadget is _not_ running remotely
		metricsPrint := annotations[AnnotationMetricsPrint] == "true" && !gadgetCtx.IsRemoteCall()

		// Not collecting AND not printing, do nothign for this data source
		if !metricsCollect && !metricsPrint {
			continue
		}

		// we only use the global instance if there's an explicit name mapping available; otherwise we'll fallback
		// to using a dedicated local registry/collector instance
		useGlobal := false
		if metricsPrint {
			gadgetCtx.Logger().Debugf("enabling print for %s", ds.Name())

			// Disable original data source to avoid other operators subscribing to it
			ds.SetRequested(false)

			// Create a new data source for the output with a single field
			odsName := fmt.Sprintf("%s-%s", ds.Name(), PrintDataSourceSuffix)
			ods, err := gadgetCtx.RegisterDataSource(datasource.TypeSingle, odsName)
			if err != nil {
				return fmt.Errorf("registering %q: %w", odsName, err)
			}

			// set default annotations
			ods.AddAnnotation("cli.supported-output-modes", "raw")
			ods.AddAnnotation("cli.default-output-mode", "raw")
			ods.AddAnnotation("cli.clear-screen-before", "true")

			// Use annotations from the original data source
			for k, v := range ds.Annotations() {
				ods.AddAnnotation(k, v)
			}

			f, err := ods.AddField(PrintFieldName, api.Kind_String, datasource.WithAnnotations(map[string]string{"content-type": "text/plain"}))
			if err != nil {
				return fmt.Errorf("adding field %q: %w", PrintFieldName, err)
			}

			// TODO: Store these fields per datasource to support multiple datasources
			m.outputDS = ods
			m.outputField = f
		}

		// if there's an explicit mapping set for the datasource, we will export it using the global exporter
		// (if that is available)
		mappedName, ok := m.nameMappings[ds.Name()]
		if ok && m.op.exporter != nil {
			useGlobal = true
		} else if ok {
			gadgetCtx.Logger().Warnf("global exporter not configured, using local metric instance")
		}

		// If mapped name is empty, it hasn't been explicitly set (or set to empty), so we will use the data source name
		if mappedName == "" {
			mappedName = ds.Name()
		}

		gadgetCtx.Logger().Debugf("collecting metrics for data source %q as %q", ds.Name(), mappedName)

		m.collectors[ds] = &metricsCollector{
			output:            metricsPrint,
			mappedName:        mappedName,
			useGlobalProvider: useGlobal,
		}
	}
	return nil
}
```

Reply:
It's and it's clearer. So, using it.